pkg/api/pkgecosystem/ecosystem.go

```diff
@@ -71,3 +77,16 @@ func EcosystemsAsStrings(es []Ecosystem) []string {
 	}
 	return s
 }
+
+// Parse returns an Ecosystem corresponding to the given string name, or
+// the None ecosystem along with an error if there is no matching Ecosystem.
+// If name == "", then the None ecosystem is returned with no error.
+func Parse(name string) (Ecosystem, error) {
+	for _, s := range append(SupportedEcosystems, None) {
+		if string(s) == name {
```

Comment:
Do the PURL type names all match our internal ecosystem naming? e.g. crates.io looks different from https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#cargo

Reply:
Ah good catch. I need to fix the downloader for that

Reply:
Done